### PR TITLE
proposals: add rkt

### PR DESCRIPTION
**Pull request by jonboulle**

### CODE REVIEW

File: proposals/rkt.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+*Source control repositories*:
+
+* https://github.com/coreos/rkt (hoping to move to github.com/rkt/rkt, pending availability of that GH organisation)
+
+*Initial Committers*:
+
+Core: @alban, @philips, @squeed, @dgonyeo, @euank, @iaguis, @jonboulle, @krnowak, @lucab, @s-urbaniak, @steveeJ, @yifan-gu
+
+*Infrastructure requirements*: CI might be useful; currently CI is sponsored by CoreOS <https://jenkins-rkt-public.prod.coreos.systems/#>
```
Comment (caniszczyk) [inline proposals/rkt.adoc]: any particular reason you use your own hosted Jenkins vs say Travis CI or other systems out there?

Comment (jonboulle) [inline proposals/rkt.adoc]: Yes, because rkt is complicated :-). Since it's very invasive to a system, we want to do tests in VMs; further, we test on different OS platforms (e.g. Debian/Fedora/etc) and hence spin up EC2 instances for each run.

**Comment by caniszczyk** [comment]
RFC @cncf/toc

File: proposals/rkt.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
+## rkt
+
+*Name of project*: rkt
+
+*Description*:
+
+rkt is an application container engine developed for modern production cloud-native environments. It features a pod-native approach, a pluggable execution architecture, and a well-defined surface area that makes it ideal for integration with other systems.
+
+The core execution unit of rkt is the _pod_, a collection of one or more applications executing in a shared context. (rkt's pods are synoymous with the concept in the Kubernetes orchestration system). rkt allows users to apply different configurations (like isolation parameters) at both pod-level and at the more granular per-application level. rkt's architecture means that each pod executes directly in the classic Unix process model (i.e. there is no central daemon), in a self-contained, isolated environment.
```
Comment (rothgar) [inline proposals/rkt.adoc]: "self-contained, isolated environment" a.k.a. containerized?
Other proposals use the phrase "containerized applications" so language should stay consistent. I'm also not sure I understand what "self-contained" means for an application.

File: proposals/rkt.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+*Release methodology and mechanics*: rkt has a mature and regular release process that is [documented and automated](https://github.com/coreos/rkt/blob/4573a6bb93945ddb12891c667f57f650f867f9b3/Documentation/devel/release.md). It is available for most major Linux distributions and every rkt release builds self-contained RPM/deb packages that users can install. Further, these packages are available as part of the Kubernetes repos to enable testing of the rkt + Kubernetes integration. 
+
+*Social media accounts*: None
+
+*Existing sponsorship*: CoreOS
+
+**Orchestration Integrations**
+
+* [Kubernetes](https://kubernetes.io/docs/getting-started-guides/rkt/): an enterprise-grade container orchestration solution. Kubernetes offers rkt support, including via [minikube](https://github.com/kubernetes/minikube#using-rkt-container-engine) which makes it easy to test and run locally. The new rkt integration via Kubernetes's CRI (Container Runtime Interface) is currently passing 84/89 conformance tests; the hope would be to get this to 100% as Kubernetes 1.7 work begins and CRI stabilizes to beta.
```
Comment (rothgar) [inline proposals/rkt.adoc]: Just out of curiosity, what are the 5 failing tests?

File: proposals/rkt.adoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+A pillar of Cloud Native is packaging and distributing applications as container images, and a container runtime is hence a critical building block for Cloud Native computing. rkt provides a tightly-scoped tool for this purpose, leading to various integrations in Cloud Native orchestration systems like Kubernetes, Mesos, Nomad, and many organization's bespoke systems.
+
+The rkt project has a proven track record of innovation and open source collaboration. The container engine has differentiated itself via a pod-native approach to process management, which integrates into existing management paradigms; reuse of existing components whenever possible; and a focus on simplicity. Many projects, individuals, and organizations have chosen rkt to run their container-packaged applications, both alone and through its integration with Kubernetes. All of this is directly inline with the CNCF's container-packaged mission.
+
+*External Dependencies*:
+
+The rkt project uses [glide](https://github.com/Masterminds/glide) to maintain dependencie - an up-to-date manifest can be found in the [glide.lock file in the repository](https://github.com/coreos/rkt/blob/master/glide.lock).
```
Comment (rothgar) [inline proposals/rkt.adoc]: dependencie - missed an 's'

File: proposals/rkt.adoc (hunk 3 above)
Comment (jonboulle) [inline proposals/rkt.adoc]: https://github.com/kubernetes-incubator/rktlet/issues/95#issuecomment-278346394 (but likely needs an update since then)

File: proposals/rkt.adoc (hunk 4 above)
Comment (jonboulle) [inline proposals/rkt.adoc]: thanks, fixed

File: proposals/rkt.adoc (hunk 2 above)
Comment (jonboulle) [inline proposals/rkt.adoc]: This is trying to explain that the pod is self-contained, not just the application. Contrast to, say, a pod assembled from multipleindividual runc or systemd-nspawn containers. Open to other wording suggestions here?

Comment (rothgar) [inline proposals/rkt.adoc]: This probably doesn't matter for this proposal but for the sake of clarification...
I'm not a fan of "pod-native" because it's meaningless without a background and understanding of what a pod means in Kubernetes. Even then, many people who run Kubernetes I don't use multi-container deployments and just think of a pod as a container.

I would probably reword it to say something like multi-container application deployments or multi-container bundles. Something that's more descriptive and not as abstract.

I'm sure everyone reading this proposal knows exactly what is meant by a "pod" but that term caused confusion for the other descriptions of rkt.

Comment (jonboulle) [inline proposals/rkt.adoc]: Hm, this paragraph does lead with an explanation of what pods are. I've added a link to the Kubernetes pod definition too. Does that help? (I honestly really dislike the pod/container conflation so don't want to repeat it here)

Comment (rothgar) [inline proposals/rkt.adoc]: Yes I like the explanation + link. Thanks. :clap:

**Comment by caniszczyk** [comment]
final RFC on comments @cncf/toc, voting will happen Friday

**Comment by caniszczyk** [comment]
FYI @cncf/toc rkt vote is out: https://lists.cncf.io/pipermail/cncf-toc/2017-March/000741.html

File: proposals/rkt.adoc (hunk 1 above)
Comment (lucab) [inline proposals/rkt.adoc]: For reference, we use travis-ci + semaphore-ci + jenkins as they cover different testing scenarios. Only the latter is on private infrastructure though.

File: proposals/rkt.adoc (hunk 3 above)
Comment (lucab) [inline proposals/rkt.adoc]: As per the above report, the figure is actually "passing 89/94".

Comment (jonboulle) [inline proposals/rkt.adoc]: thanks, fixed

**Comment by caniszczyk** [comment]
rkt passes with 8/9 +1 binding @cncf/toc votes:

https://lists.cncf.io/pipermail/cncf-toc/2017-March/000778.html

Thanks everyone who helped with the proposal and welcome rkt community to the CNCF!
